src/reputation/tasks.py

```diff
@@ -126,16 +173,49 @@ def check_transaction_success(transaction_hash):
     queue=QUEUE_PURCHASES,
 )
 def check_deposits():
-    deposits = Deposit.objects.filter(paid_status=None)
+    # Sort by created date to ensure a malicious user doesn't attempt to take credit for
+    # a deposit made by another user. This is a temporary solution until we add signed messages
+    # to validate users own wallets.
+    deposits = Deposit.objects.filter(paid_status=None).order_by("created_date")
+
     for deposit in deposits.iterator():
-        amt = deposit.amount
+        # If a deposit is not resolved after 6 hours, mark it as failed
```

Comment:
I wonder if we need to reduce the time here. 6 hours seems like a lot?

Reply:
Set to 1 hour for now.

Reply:
Yeah 1 hour seems good